Classes/Command/UpdateInheritanceCommand.php

```diff
@@ -0,0 +1,301 @@
+        }
+
+        $products = $this->fetchInheritanceQueue((int)$input->getOption('limit'));
+        if (empty($products)) {
+            $io->success('No products in queue, no processing needed.');
+
+            return true;
+        }
+
+        $queueCount = $this->fetchInheritanceQueueCount();
+        $io->section(
+            'Check inheritance status for ' . count($products) . ' of ' . $queueCount . ' product(s) total in queue.'
+        );
+
+        $updated = 0;
+        foreach ($products as $index => $product) {
+            $io->section(
+                sprintf(
+                    'Check inheritance for product [%s] (%s/%s)',
+                    $product['product_uid'],
+                    $index + 1,
+                    count($products)
+                )
+            );
+
+            $inheritdData = DataInheritanceUtility::inheritDataFromParent($product['product_uid']);
+
+            if (!empty($inheritdData)) {
+                $dataHandler = GeneralUtility::makeInstance(DataHandler::class);
+                $dataHandler->start($inheritdData['data'], $inheritdData['cmd']);
+                $dataHandler->process_datamap();
+                $dataHandler->process_cmdmap();
+            }
+
+            if (
+                !empty($inheritdData)
+                && isset($inheritdData['data'][ProductRepository::TABLE_NAME])
+            ) {
+                foreach ($inheritdData['data'][ProductRepository::TABLE_NAME] as $id => $data) {
+                    $updated++;
+                    unset($data['is_inherited']);
+                    $io->writeln(
+                        'Inherited data was updated for product: ' . $id .
+                        ', fields: ' . implode(',', array_keys($data))
+                    );
+                }
+            }
+
+            $this->removeProductFromInheritanceQueue($product['product_uid']);
+        }
+
+        $io->success(
+            sprintf(
+                'Inheritance status was checked for %s product(s) and updated on %s product(s).',
+                count($products),
+                $updated
+            )
+        );
+
+        return true;
+    }
+
+    /**
+     * Fetch all products with parent.
+     *
+     * @return array
+     */
+    protected function fetchChildProducts(): array
+    {
+        $queryBuilder = GeneralUtility::makeInstance(ConnectionPool::class)
+            ->getQueryBuilderForTable(ProductRepository::TABLE_NAME);
+        $queryBuilder->getRestrictions()->removeAll();
+
+        $records = $queryBuilder->select('uid')
+            ->from(ProductRepository::TABLE_NAME)
+            ->where(
+                $queryBuilder->expr()->eq(
+                    'deleted',
+                    $queryBuilder->createNamedParameter(0, \PDO::PARAM_INT)
+                ),
+                $queryBuilder->expr()->gt(
+                    'parent',
+                    $queryBuilder->createNamedParameter(0, \PDO::PARAM_INT)
+                )
```

Comment:
"Not equal" is probably going to be faster here as it will rely more on the index. https://stackoverflow.com/questions/4753955/mysql-equal-vs-greater-thanless-than-performance-difference